fingertip/util/filesystem.py

```diff
@@ -0,0 +1,99 @@
+                   check=True)
+
+
+def mount_supported_fs(backing_file, tgt):
+    # use warning to avoid rewriting the prompt with sudo prompt
+    log.warning('mounting a reflink-supported filesystem for image storage...')
+    tgt_uid, tgt_gid = os.stat(tgt).st_uid, os.stat(tgt).st_gid
+    subprocess.run(['sudo', 'mount', '-o', 'loop', backing_file, tgt],
+                   check=True)
+    mount_uid, mount_gid = os.stat(tgt).st_uid, os.stat(tgt).st_gid
+    if (tgt_uid, tgt_gid) != (mount_uid, mount_gid):
+        log.debug(f'fixing owner:group ({tgt_uid}:{tgt_gid})')
+        subprocess.run(['sudo', 'chown', f'{tgt_uid}:{tgt_gid}', tgt],
+                       check=True)
+
+def storage_schedule_cleanup():
+    # Do this only if fingertip is in PATH
+    if not shutil.which("fingertip"):
+        log.warning('No `fingertip` found in PATH. Not scheduling '
+                    'automatic cleanup')
+        return
+
+    subprocess.run(['systemd-run', '--user', '--on-active', '12h',
+                    'fingertip', 'cleanup', 'periodical'])
+
+
+def storage_setup_wizard():
+    assert SETUP in ('auto', 'suggest', 'never')
+    if SETUP == 'never':
+        return
+    size = SIZE
+    machinesdir = MACHINES
+    if not os.path.exists(machinesdir):
+        os.mkdir(machinesdir)
+    backing_file = os.path.join(machinesdir, 'for-machines.xfs')
+    if not is_supported(machinesdir):
+        log.warning(f'images directory {machinesdir} lacks reflink support')
+        log.warning('without it, fingertip will thrash and fill up your SSD in no time')
+        if not os.path.exists(backing_file):
+            if SETUP == 'suggest':
+                log.warning(f'would you like to allow fingertip to allocate {size} '
+                            f'at {backing_file} '
+                            'for a reflink-enabled XFS loop mount?')
+                log.warning('(set FINGERTIP_SETUP="auto" environment variable'
+                            ' to do it automatically)')
+                i = input(f'[{size}]/different size/cancel/ignore> ').strip()
+                if i == 'cancel':
+                    log.error('cancelled')
+                    sys.exit(1)
+                elif i == 'ignore':
+                    return
+                size = i or size
+            tmp = temp.disappearing_file(CACHE)
+            create_supported_fs(tmp, size)
+            os.rename(tmp, backing_file)
+        log.warning(f'fingertip will now mount the XFS image at {backing_file}')
+        if SETUP == 'suggest':
+            i = input(f'[ok]/skip/cancel> ').strip()
+            if i == 'skip':
+                log.warning('skipping; fingertip will have no reflink superpowers')
+                log.warning('tell your SSD I\'m sorry')
+                return
+            elif i and i != 'ok':
+                log.error('cancelled')
+                sys.exit(1)
+        mount_supported_fs(backing_file, MACHINES)
+
+        # Schedule automatic cleanup
```

Comment:
Why condition scheduling cleanup on creating an FS-in-a-file? I'm sure that users having a CoW-enabled FS would still like cleanup to happen.

Reply:
RIght. I mixed up the indentation.